pkg/operator/controller/canary/service_account_test.go

```diff
@@ -81,6 +81,15 @@ func Test_canaryServiceAccountChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if annotations change",
+			mutate: func(sa *corev1.ServiceAccount) {
+				sa.Annotations = map[string]string{
+					"foo": "bar",
+				}
+			},
+			expect: true,
```

Comment:
Ah, so the intended behavior is that the controller _does_ stomp annotations?  Is there a reason it should do that?